sphinx/environment/__init__.py

```diff
@@ -668,6 +669,10 @@ def read_doc(self, docname, app=None):
         self.settings['trim_footnote_reference_space'] = \
             self.config.trim_footnote_reference_space
         self.settings['gettext_compact'] = self.config.gettext_compact
+        language = (self.config.language or 'en').replace('_', '-')
+        self.settings['language_code'] = language
+        if language in smartchars.quotes:
+            self.settings['smart_quotes'] = self.config.html_use_smartypants
```

Comment:
Hmm... this causes `html_use_smartypants` affects to other builders. This method is called for all builders.
Is this intended? If so, it should be renamed to non `html_` prefix one.
If not, it would be better to use post_transform. It can be applied only if HTML builders.